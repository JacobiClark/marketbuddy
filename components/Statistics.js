import React, { useState, useEffect, Fragment } from "react";
import {
  Box,
  Flex,
  Spinner,
  Center,
  Table,
  Thead,
  Tbody,
  Tfoot,
  Tr,
  Th,
  Td,
  TableCaption,
} from "@chakra-ui/react";
import StatBox from "./StatBox";

const Statistics = ({ ticker }) => {
  const [statisticsData, setStatisticsData] = useState({});
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const fetchStatisticsData = async () => {
      try {
        const res = await fetch(
          "https://apidojo-yahoo-finance-v1.p.rapidapi.com/stock/v3/get-statistics?symbol=" +
            ticker,
          {
            method: "GET",
            headers: {
              "x-rapidapi-key": process.env.YAHOO_FINANCE_KEY,
              "x-rapidapi-host": "apidojo-yahoo-finance-v1.p.rapidapi.com",
            },
          }
        );
        const statistics = await res.json();
        setStatisticsData(statistics);
        setIsLoading(false);
      } catch (error) {
        console.error(error);
      }
    };
    fetchStatisticsData();
  }, [ticker]);

  if (isLoading) {
    return (
      <Box align="center">
        <Spinner />
      </Box>
    );
  }

  return (
    <Fragment>
      <Center>
        <Box maxW="900">
          <Table variant="simple">
            <Tbody>
              <Tr>
                <Td>Market Cap</Td>
                <Td isNumeric>
                  {statisticsData?.summaryDetail?.marketCap?.fmt ?? "--"}
                </Td>
              </Tr>
              <Tr>
                <Td>Cash On Hand</Td>
                <Td isNumeric>
                  {statisticsData?.financialData?.totalCash?.fmt ?? "--"}
                </Td>
              </Tr>
              <Tr>
                <Td>Debt</Td>
                <Td isNumeric>
                  {statisticsData?.financialData?.totalDebt?.fmt ?? "--"}
                </Td>
              </Tr>
              <Tr>
                <Td>Profit Margins</Td>
                <Td isNumeric>
                  {statisticsData?.defaultKeyStatistics?.profitMargins?.fmt ??
                    "--"}
                </Td>
              </Tr>

              <Tr>
                <Td>Last Dividend</Td>
                <Td isNumeric>
                  {statisticsData?.defaultKeyStatistics?.lastDividendValue
                    ?.fmt ?? "--"}
                </Td>
              </Tr>
              <Tr>
                <Td>Short Percentage</Td>
                <Td isNumeric>
                  {statisticsData?.defaultKeyStatistics?.shortPercentOfFloat
                    ?.fmt ?? "--"}
                </Td>
              </Tr>
              <Tr>
                <Td>Daily Volume</Td>
                <Td isNumeric>
                  {statisticsData?.price?.regularMarketVolume?.fmt ?? "--"}
                </Td>
              </Tr>
            </Tbody>
          </Table>
        </Box>
      </Center>
    </Fragment>
  );
};

export default Statistics;
