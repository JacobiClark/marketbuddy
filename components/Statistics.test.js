import React from "react";
import renderer from "react-test-renderer";

import Statistics from "./Statistics";

it("Renders correctly without items", () => {
  const tree = renderer.create(<Statistics />).toJSON();
  expect(tree).toMatchSnapshot();
});
