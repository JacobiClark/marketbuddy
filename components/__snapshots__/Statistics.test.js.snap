// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly without items 1`] = `
<div
  align="center"
  className="css-0"
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1y7joxr{display:inline-block;border-color:currentColor;border-style:solid;border-radius:99999px;border-width:2px;border-bottom-color:transparent;border-left-color:transparent;-webkit-animation:animation-b7n1on 0.45s linear infinite;animation:animation-b7n1on 0.45s linear infinite;}@-webkit-keyframes animation-b7n1on{0%{-webkit-transform:rotate(0deg);-moz-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}@keyframes animation-b7n1on{0%{-webkit-transform:rotate(0deg);-moz-transform:rotate(0deg);-ms-transform:rotate(0deg);transform:rotate(0deg);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}",
      }
    }
    data-emotion="css 1y7joxr animation-b7n1on"
  />
  <div
    className="chakra-spinner css-1y7joxr"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-f8n5zr{border:0px;clip:rect(0px, 0px, 0px, 0px);height:1px;width:1px;margin:-1px;padding:0px;overflow:hidden;white-space:nowrap;position:absolute;}",
        }
      }
      data-emotion="css f8n5zr"
    />
    <span
      className="css-f8n5zr"
    >
      Loading...
    </span>
  </div>
</div>
`;
